include/rocksdb/compaction_filter.h

```diff
@@ -13,24 +13,16 @@
 #include <string>
 #include <vector>
 
+#include "rocksdb/listener.h"
```

Comment:
Rather than including listener, can we include "types.h" and move the enum you are using into there?  The incestual nature of the includes makes the builds take forever...

Reply:
Good point, done.